File: Solutions/1422/1422.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/8/14 8:45
# File: 1422.py
# Desc: 


class Solution:
    def maxScore(self, s: str) -> int:
        z = 0 if s[0] == '1' else 1
        y = s[1:].count('1')
        ret = z+y
        for i in s[1:-1]:
            if i == '0':
                z += 1
            else:
                y -= 1
            ret = max(ret, z+y)
        return ret

File: Solutions/1422/1422.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1422.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-8-15 10:43:01
***************************************************************************************************/


int maxScore(char * s){
    int z, y = 0;
    z = s[0] == '0' ? 1 : 0;
    for (int i=1; i<strlen(s); i++) {
        if (s[i] == '1') y++;
    }
    int max = z + y;
    for (int i=1; i<strlen(s)-1; i++) {
        if (s[i] == '0') z++;
        else y--;
        max = max > z+y ? max : z+y;
    }
    return max;
}
